pulpcore/app/models/publication.py

```diff
@@ -456,6 +460,54 @@ class Meta:
         )
 
 
+class HeaderContentGuard(ContentGuard, AutoAddObjPermsMixin):
+    """
+    Content guard to protect content based on a header value.
+    """
+
+    TYPE = "header"
+
+    header_name = models.TextField()
+    json_accessor = models.TextField(null=True)
+    value = models.TextField()
+
+    def permit(self, request):
+        header_content = request.headers.get(self.header_name)
+        if not header_content:
+            raise PermissionError("Access not allowed. Header not found.")
```

Comment:
We should just access deny here but log (debug?) a bit more verbose.
```suggestion
            raise PermissionError("Access denied.")
            log.debug(_("Access denied. Header {header_name} not found in request.").format(header_name))
```
The same with all raise below.